### PR TITLE
[next] Extras - new Gizmo using Three Viewport Gizmo library

### PR DESCRIPTION
**Pull request by michealparks**
### Overview

This PR overhauls and simplifies our Gizmo by letting `three-viewport-gizmo` do a lot of the heavy work. Replacing it solves a few issues and adds many new features:
- Adds `camera-controls` support
- Fixes a few few rendering bugs with the last implementation
- Adds alternate coordinate system support.
- Adds an alternative box gizmo.
- [A bunch of new options](https://fennec-hub.github.io/three-viewport-gizmo/api.html) from the library itself.

I've updated the docs accordingly.

### CODE REVIEW

File: packages/extras/src/lib/components/controls/OrbitControls/OrbitControls.svelte [hunk 1]
```diff
@@ -31,17 +31,25 @@
...
+      props.onchange?.(event)
+    }
+
+    orbitControls.set(controls)
+    controls.addEventListener('change', handleChange)
+    return () => {
+      orbitControls.set(undefined)
+      controls.removeEventListener('change', handleChange)
+    }
+  })
```
Comment (michealparks) [inline packages/extras/src/lib/components/controls/OrbitControls/OrbitControls.svelte:47]: This fixes a bug in which OrbitControls was overwriting `onchange` listeners in some circumstances (like the gizmo example)

Comment (joshwashywash) [reply packages/extras/src/lib/components/controls/OrbitControls/OrbitControls.svelte:47]: i think i have the same sort of fix in my local experimenting branch for #1237

File: packages/extras/src/lib/components/Gizmo/Gizmo.svelte [hunk 2]
```diff
@@ -1,467 +1,143 @@
...
-      clickTarget.style.bottom = ''
-      clickTarget.style.top = `${boundingRect.bottom - size - paddingY}px`
-    } else {
-      clickTarget.style.bottom = ''
-      clickTarget.style.top = `${paddingY + boundingRect.top}px`
-    }
-
-    clickTarget.style.height = `${size}px`
-    clickTarget.style.width = `${size}px`
+    ref.camera = $camera as PerspectiveCamera | OrthographicCamera
```
Comment (joshwashywash) [inline packages/extras/src/lib/components/Gizmo/Gizmo.svelte:63]: so what i've been doing in some components is allow an optional camera prop that if not present will default to the camera in `useThrelte` something like this

```svelte
<script lang="ts">
  type Props = { camera: PerspectiveCamera | OrthographicCamera }
  let { camera, ref } = props()
  
  const { camera: defaultCamera } = useThrelte()
  
  // can't remember if defaultCamera is a store or has a `current` prop
  const _camera = $derived(camera ?? $defaultCamera)
  ref = new ViewportGizmo(camera.current)
  $effect(() => {
    if (_camera !== ref.camera) {
      ref.camera = _camera
    }
  })
</script>
```

should we do the same here? i don't think you'd ever want to control a camera that wasn't the one used for rendering but maybe???

File: packages/extras/src/lib/components/Gizmo/Gizmo.svelte [hunk 3]
```diff
@@ -1,467 +1,143 @@
...
+    if (!_controls) return
 
-    if (intersects.length > 0) {
-      const alreadyReached = handleIntersection(intersects[0])
-      if (alreadyReached) {
-        // get the second closest intersection
-        if (intersects.length > 1) {
-          handleIntersection(intersects[1])
-        }
+    const handleStart = () => (_controls.enabled = false)
```
Comment (joshwashywash) [inline packages/extras/src/lib/components/Gizmo/Gizmo.svelte:69]: it looks like attaching the gizmo will do this already i think https://github.com/Fennec-hub/three-viewport-gizmo/blob/291992c3a03648acf9ac7318a9c38ad7e4c797d6/lib/ViewportGizmo.ts#L377-L381

Comment (michealparks) [reply packages/extras/src/lib/components/Gizmo/Gizmo.svelte]: good catch, fixed up!

File: packages/extras/src/lib/components/Gizmo/Gizmo.svelte (hunk 2 above)
Comment (michealparks) [reply packages/extras/src/lib/components/Gizmo/Gizmo.svelte:63]: camera arg added!

**Pull request merged**